app/src/main/java/com/example/cartoon/Notification.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.cartoon
+import android.Manifest
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.content.pm.PackageManager
+import android.graphics.Color
+import android.os.Build
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.shashank.sony.fancytoastlib.FancyToast
+
+class Notification(private val context: Context) {
+
+    private val CHANNEL_ID = "channelId"
+    private val CHANNEL_NAME = "channelName"
+
+    init {
+        createNotificationChannel()
+    }
+
+    fun sendNotification(title: String, message: String, smallIcon: Int, notificationId: Int) {
+        val notification = NotificationCompat.Builder(context, CHANNEL_ID)
+            .setContentTitle(title)
+            .setContentText(message)
+            .setSmallIcon(smallIcon)
+            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+            .build()
+
+        val notificationManager = NotificationManagerCompat.from(context)
+        if (ActivityCompat.checkSelfPermission(
+                context,
+                Manifest.permission.POST_NOTIFICATIONS
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            FancyToast.makeText(
+                context,
+                "No image selected", FancyToast.LENGTH_LONG,
+                FancyToast.WARNING,true).show()
+
+        }
+        notificationManager.notify(notificationId, notification)
```

Comment:
Hey @Maan-Sharma . You should check the notification permission in activity itself before calling sendNotification. Reason being. 
1. check the notification permission -> register activity for result - > you will know whether user have given permission or not. Take action based on granted or not granted. 


You can refer this. https://developer.android.com/training/permissions/requesting

Reply:
@ashut0sh75 . I think this change needed here. Can cause a crash in android 13 and above. due to the notification being triggered without user permission.

Reply:
Thanks for pointing it out @lingarajsankaravelu . I tried the app in 13 and we saw that notification is not generated also I think there was an issue with toast being triggered unnecessarily. 
Should we ask for users permissions separately to resolve it ?

Reply:
@ashut0sh75 . Yes users permission should be from activity.

Reply:
okk sure.